lsp/nls/src/linearization/completed.rs

```diff
@@ -40,14 +34,12 @@ impl Completed {
 
     pub fn get_in_scope(
         &self,
-        LinearizationItem { scope, .. }: &LinearizationItem<Resolved>,
+        LinearizationItem { env, .. }: &LinearizationItem<Resolved>,
     ) -> Vec<&LinearizationItem<Resolved>> {
-        let empty = Vec::with_capacity(0);
-        (0..scope.len())
-            .flat_map(|end| self.scope.get(&scope[..=end]).unwrap_or(&empty))
-            .map(|id| self.get_item(*id))
-            .flatten()
-            .collect()
+        env.iter()
+            .map(|(_, id)| self.get_item(*id))
+            .collect::<Option<Vec<_>>>()
+            .unwrap_or_default()
```

Comment:
Just want to make sure I'm understanding this correctly: we iterate through every term in the environment, and then find each element in the linearization, but if even a single one is missing, we return an empty vec?

Am I right in thinking that missing terms would have to be the result of an error in the construction of the linearization?

Reply:
Wow, nice catch. What I want to do is accumulate the `Some(..)` just like `sequenceA` in Haskell. But I just tested it and it seems this one returns `None` if there is any `None` present in the list.

Reply:
Is there a function to do this or do I have to implement it manually?

Reply:
Ah, in that case you probably want [`filter_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map).

Reply:
If the linearization MUST have an item at an index, I tend to think we should panic. Otherwise, it may let us slip bugs silently.

Reply:
Yeah that’s true, because the environment cannot/shouldn’t have an id that is not in the linearization.